pkg/shed/metrics.go

```diff
@@ -13,95 +13,193 @@ type metrics struct {
 	// all metrics fields must be exported
 	// to be able to return them by Metrics()
 	// using reflection
-	PutCounter            prometheus.Counter
-	PutFailCounter        prometheus.Counter
-	GetCounter            prometheus.Counter
-	GetFailCounter        prometheus.Counter
-	GetNotFoundCounter    prometheus.Counter
-	HasCounter            prometheus.Counter
-	HasFailCounter        prometheus.Counter
-	DeleteCounter         prometheus.Counter
-	DeleteFailCounter     prometheus.Counter
-	IteratorCounter       prometheus.Counter
-	WriteBatchCounter     prometheus.Counter
-	WriteBatchFailCounter prometheus.Counter
+	DBOpenCount          prometheus.Counter
+	GetCount             prometheus.Counter
+	GetFailCount         prometheus.Counter
+	GetNotFoundCount     prometheus.Counter
+	PutCount             prometheus.Counter
+	PutFailCount         prometheus.Counter
+	HasCount             prometheus.Counter
+	HasFailCount         prometheus.Counter
+	DeleteCount          prometheus.Counter
+	DeleteFailCount      prometheus.Counter
+	TotalCount           prometheus.Counter
+	TotalFailCount       prometheus.Counter
+	CountPrefixCount     prometheus.Counter
+	CountPrefixFailCount prometheus.Counter
+	CountFromCount       prometheus.Counter
+	CountFromFailCount   prometheus.Counter
+	IterationCount       prometheus.Counter
+	IterationFailCount   prometheus.Counter
+	FirstCount           prometheus.Counter
+	FirstFailCount       prometheus.Counter
+	LastCount            prometheus.Counter
+	LastFailCount        prometheus.Counter
+	GetBatchCount        prometheus.Counter
+	WriteBatchCount      prometheus.Counter
+	WriteBatchFailCount  prometheus.Counter
+	DBCloseCount         prometheus.Counter
 }
 
 func newMetrics() metrics {
 	subsystem := "shed"
 
 	return metrics{
-		PutCounter: prometheus.NewCounter(prometheus.CounterOpts{
+		DBOpenCount: prometheus.NewCounter(prometheus.CounterOpts{
 			Namespace: m.Namespace,
 			Subsystem: subsystem,
-			Name:      "put_count",
-			Help:      "Number of times the PUT operation is done.",
+			Name:      "db_db_open_count",
```

Comment:
Every metric name now has a prefix `db_`, this is not needed as there already is a subsystem "shed" set. It would be nicer to remove `db_` prefix for every metric in this PR.